src/db/migrations/20240320074757_initial_migration.ts

```diff
@@ -10,23 +10,29 @@ export async function up(knex: Knex): Promise<void> {
         table.text("hashed_password").notNullable();
         table.timestamp("created_at").defaultTo(knex.fn.now());
         // TODO: Implement user roles (e.g., admin, user, etc.)
+        table.enum("permission_level", ["admin", "user"]).defaultTo("user");
         table.text("school");
         table.text("name");
         table.text("country");
+        table.index("username");
+        table.index("email");
       })
       .createTable("tasks", (table) => {
         table.uuid("id").primary().defaultTo(knex.raw("uuid_generate_v4()"));
         table.text("slug").notNullable().unique();
         table.text("title").notNullable();
         table.text("description");
         table.text("statement").notNullable();
-        table.decimal("score_max", 10, 4).notNullable();
+        table.decimal("score_max", 10, 4).notNullable().defaultTo(100.0);
         table.text("mvp_output"); // Delete this later. It's just for making progress
+        table.float("time_limit").checkPositive("tasks_time_limit_checkPositive").defaultTo(2.0); // in seconds
+        table.integer("memory_limit").unsigned().defaultTo(100000); // in bytes
```

Comment:
There's no unsigned int in postgres. Also please remove the default.

We'd rather have a failed task creation than a mistaken one.

Mistake => bad contest
Failed save => edit and try again before contest even starts

Reply:
Also, when you have things like these where the units might be unclear, I recommend putting the units in the name (not in comments).

See: `running_time_ms` very clear to everyone that units is in ms. Frontend guys, backend guys, people editing the schema, people romping around in SQL. Comments only exist in this one line in this one file.